jspwiki-bootstrap/src/main/java/org/apache/wiki/bootstrap/WikiBootstrapServletContextListener.java

```diff
@@ -98,4 +101,33 @@ ConfigurationSource createConfigurationSource( final Properties properties ) {
     public void contextDestroyed( final ServletContextEvent sce ) {
     }
 
+    /**
+     * This method sets the JSPWiki working directory (jspwiki.workDir). It first checks if this property
+     * is already set. If it isn't, it attempts to use the servlet container's temporary directory
+     * (javax.servlet.context.tempdir). If that is also unavailable, it defaults to the system's temporary
+     * directory (java.io.tmpdir).
+     * <p>
+     * This method is package-private to allow for unit testing.
+     *
+     * @param properties     the JSPWiki properties
+     * @param servletContext the Servlet context from which to fetch the tempdir if needed
+     * @since JSPWiki 2.11.1
+     */
+    void setWorkDir(final Properties properties, final ServletContext servletContext) {
```

Comment:
Hi,

my bad, I was meaning do all of these changes inside `PropertyReader`, just before the [variable expansion](https://github.com/apache/jspwiki/blob/master/jspwiki-util/src/main/java/org/apache/wiki/util/PropertyReader.java#L148). That way all property reading/mungling is done inside `PropertyReader#loadWebAppProps`

Reply:
done.